backend/controller/AccountController.ts

```diff
@@ -0,0 +1,207 @@
+        if (usernameOrEmail.includes('@')) {
+            users = await DBUser.findBy({
+                mail: usernameOrEmail,
+            });
+        } else {
+            users = await DBUser.findBy({
+                name: usernameOrEmail,
+            });
+        }
+        if (users.length > 1) {
+            throw new LoernwerkError(
+                'ambiguous user details',
+                LoernwerkErrorCodes.AMBIGUOUS_INFORMATION
+            );
+        } else if (users.length === 0) {
+            throw new LoernwerkError(
+                'Mail/Name not matching an existing User',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        } else {
+            const founduser = users[0];
+            if (await this.validatePassword(password, founduser.password)) {
+                return founduser;
+            }
+            throw new LoernwerkError(
+                'Incorrect Password',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+    }
+
+    /**
+     * Searches for an Account with corresponding Id in the database and returns it. Throws an error if no account was found.
+     * @param id the id of the account
+     * @returns the found user
+     */
+    public static async getAccountById(id: number): Promise<IUser> {
+        const user = await DBUser.findOneBy({ id: id });
+        if (user === null) {
+            throw new LoernwerkError(
+                'No existing User with given ID',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        return user;
+    }
+    /**
+     * Returns all users of the database in an array in reduced form.
+     * @returns A reduced version of all accounts
+     */
+    public static async getAllAccounts(): Promise<Partial<IUser>[]> {
+        return await DBUser.find({ select: { name: true, id: true } });
+    }
+
+    /**
+     * Deletes an account from the database. Throws an error if no account was found to delete was found
+     * @param id the id of the account
+     */
+    public static async deleteAccount(id: number): Promise<void> {
+        const user = await DBUser.findOneBy({ id: id });
+        if (user === null) {
+            throw new LoernwerkError(
+                'No existing User with given ID',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        await user.remove();
+    }
+
+    /**
+     * Ensures that an admin exists and creates one if none exists. This function is called called by loernwerkServer.main at server startup.
+     * The Password is printed on the console
+     */
+    public static async ensureAdminAccount(): Promise<void> {
+        const user = await DBUser.findOneBy({ type: UserClass.ADMIN });
+        if (user === null) {
+            const adminUser: Partial<IUser> = {};
+            adminUser.name = 'admin';
+            adminUser.mail = 'admin@loernwerk.de';
+            const pw = crypto.randomBytes(16).toString('hex');
```

Comment:
Is the password strong enough? I mean it has a length of 32, but only the following characters : 0123456789ABCDEF

Reply:
I do believe so, even if not, it's just a default, which can and should be changed

Reply:
It should be changed in any case and I guess no one likes a 32 length password, so i close this conversation